src/lib/planviz/ast_visualizer.cpp

```diff
@@ -13,28 +13,43 @@
 #include "optimizer/table_statistics.hpp"
 #include "sql/sql_query_plan.hpp"
 #include "utils/assert.hpp"
+#include "viz_utils.hpp"
 
 namespace opossum {
 
-void ASTVisualizer::visualize(const std::vector<std::shared_ptr<AbstractASTNode>>& ast_roots,
-                              const std::string& dot_filename, const std::string& img_filename) {
+ASTVisualizer::ASTVisualizer(const std::vector<std::shared_ptr<AbstractASTNode>>& _ast_roots,
+                             const std::string& _output_prefix, DotConfig _config)
+    : _ast_roots(_ast_roots), _output_prefix(_output_prefix), _config(_config) {}
+
+void ASTVisualizer::visualize() {
+  const auto dot_filename = _output_prefix + ".dot";
+
+  std::string img_filename = _output_prefix + ".";
+  if (_config.render_format == DotRenderFormat::PNG) {
+    img_filename += "png";
+  } else if (_config.render_format == DotRenderFormat::SVG) {
+    img_filename += "svg";
+  } else {
+    Fail("Unsupported format");
+  }
+
   // Step 1: Generate graphviz dot file
   std::ofstream file;
-  file.open(dot_filename);
+  file.open(_output_prefix + ".dot");
   file << "digraph {" << std::endl;
   file << "rankdir=BT" << std::endl;
-  file << "bgcolor=transparent" << std::endl;
+  file << "bgcolor=" << dot_color_to_string.at(_config.background_color) << std::endl;
   file << "ratio=0.5" << std::endl;
   file << "node [color=white,fontcolor=white,shape=parallelogram]" << std::endl;
   file << "edge [color=white,fontcolor=white]" << std::endl;
-  for (const auto& root : ast_roots) {
+  for (const auto& root : _ast_roots) {
     _visualize_subtree(root, file);
   }
   file << "}" << std::endl;
   file.close();
 
   // Step 2: Generate png from dot file
-  auto cmd = std::string("dot -Tpng " + dot_filename + " > ") + img_filename;
+  auto cmd = std::string("dot -Tsvg " + dot_filename + " > ") + img_filename;
```

Comment:
This does not match the comment above. Also, does this still work with iTerm?

Reply:
fixed, PNG remains default.